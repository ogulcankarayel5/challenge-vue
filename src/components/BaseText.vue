<template>
  <component :style="styles" :is="tag" class="text" :class="textClass">{{
    text
  }}</component>
</template>

<script>
export default {
  name: "BaseText",
  props: {
    tag: {
      type: String,
      default: "p",
    },
    text: {
      type: String,
    },
    size: {
      type: String,
      default: "normal",
      validator(value) {
        return ["normal", "small", "large"].indexOf(value) !== -1;
      },
    },
    bold: {
      type: Boolean,
      default: false,
    },
    color: {
      type: String,
    },
  },
  computed: {
    textClass() {
      return {
        "text-normal": this.size === "normal",
        "text-small": this.size === "small",
        "text-large": this.size === "large",

        "text-bold": this.bold,
      };
    },
    styles() {
      return {
        color: this.color,
      };
    },
  },
};
</script>

<style scoped lang="sass">
.text
    font-weight: normal
    &-normal
        font-size: 1.5rem
    &-small
        font-size: 1rem
    &-large
        font-size: 2.5rem

    &-bold
        font-weight: bold
</style>
