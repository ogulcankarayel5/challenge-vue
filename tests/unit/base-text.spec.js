import { shallowMount } from "@vue/test-utils";
import BaseText from "@/components/BaseText.vue";

const factory = (values = {}, props = {}) => {
  return shallowMount(BaseText, {
    data() {
      return {
        ...values,
      };
    },
    props: { ...props },
  });
};

describe("BaseText.vue", () => {
  it("snapshot", () => {
    const wrapper = factory();
    expect(wrapper.element).toMatchSnapshot();
  })
  it("renders text correctly ", () => {
    const msg = "Render Text";
    const wrapper = factory({}, { text: msg });
    expect(wrapper.find("p").text()).toEqual(msg);
  });
  it("renders size correctly ", () => {
    const wrapper = factory({}, { size: "small" });
    expect(wrapper.find("p").classes()).toContain("text-small");
  });
  it("renders fontWeight class correctly ", () => {
    const wrapper = factory({}, { bold: true });
    expect(wrapper.find("p").classes()).toContain("text-bold");
  });
});
