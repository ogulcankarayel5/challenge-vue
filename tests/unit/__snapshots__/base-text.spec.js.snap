// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseText.vue snapshot 1`] = `
<p
  class="text text-normal"
>
  
</p>
`;
